databuilder/databuilder/task/neo4j_staleness_removal_task.py

```diff
@@ -124,10 +130,10 @@ def validate(self) -> None:
 
     def _delete_stale_nodes(self) -> None:
         statement = textwrap.dedent("""
-        MATCH (n:{{type}})
-        WHERE {}
-        WITH n LIMIT $batch_size
-        DETACH DELETE (n)
+        MATCH (target:{{type}})
+        WHERE {{extra_condition}} {{optional_and}} {}
```

Comment:
could we give the `{}` that was there a name also. Seems a bit asymmetric to leave it out. Perhaps {{staleness_condition}}. 

Also could we reorder as `{{staleness_condition}} {{extra_condition}}`. 

And also I'm rethinking the `{{optional_and}}` part. Probably the more straightforward thing is just, when adding the extra_condition, do it like:

`statement.format(extra_condition='AND ' + extra_condition)`